src/psyclone/tests/parse/kernel_test.py

```diff
@@ -0,0 +1,272 @@
+
+# class KernelProcedure() tests
+
+
+def test_kernelprocedure_repr():
+    '''Test that the __repr__ method in KernelProcedure() behaves as
+    expected.
+
+    '''
+    tmp = create_kernelprocedure(CODE)
+    assert repr(tmp) == ("KernelProcedure(test_code)")
+
+
+def test_kernelprocedure_str():
+    '''Test that the __str__ method in KernelProcedure() behaves as
+    expected.
+
+    '''
+    tmp = create_kernelprocedure(CODE)
+    assert str(tmp) == ("    SUBROUTINE test_code()\n"
+                        "    END SUBROUTINE test_code")
+
+# class KernelType():getkerneldescriptors tests
+
+
+def test_kerneltype_typename():
+    '''Test that an exception is raised if the metadata type name is not
+    what was expected.
+
+    '''
+    my_code = CODE.replace("meta_args", "invalid_name")
+    parse_tree = parse(my_code)
+
+    with pytest.raises(ParseError) as excinfo:
+        _ = KernelType(parse_tree)
+    assert "No kernel metadata with type name 'meta_args' found." \
+        in str(excinfo.value)
+
+
+def test_kerneltype_array():
+    '''Test that an exception is raised if the metadata variable is not an
+    array.
+
+    '''
+    my_code = CODE.replace(", dimension(1)", "")
+    parse_tree = parse(my_code)
+
+    with pytest.raises(ParseError) as excinfo:
+        _ = KernelType(parse_tree)
+    assert ("In kernel metadata 'test_type': 'meta_args' variable must be "
+            "an array.") in str(excinfo.value)
+
+
+def test_kerneltype_dimensions():
+    '''Test that an exception is raised if the metadata variable is a
+    multi-dimensional array.
+
+    '''
+    my_code = CODE.replace("dimension(1)", "dimension(1,1)")
+    parse_tree = parse(my_code)
+
+    with pytest.raises(ParseError) as excinfo:
+        _ = KernelType(parse_tree)
+    assert ("In kernel metadata 'test_type': 'meta_args' variable must be a "
+            "1 dimensional array") in str(excinfo.value)
+
+
+def test_kerneltype_brackets():
+    '''Test that an exception is raised if the metadata is supplied within
+    square brackets as this is not supported within the parser.
+
+    '''
+    my_code = CODE.replace("(/", "[")
+    my_code = my_code.replace("/)", "]")
+    parse_tree = parse(my_code)
+
+    with pytest.raises(ParseError) as excinfo:
+        _ = KernelType(parse_tree)
+    assert ("Parser does not currently support [...] initialisation for "
+            "'meta_args', please use (/.../) instead.") in str(excinfo.value)
+
+
+def test_kerneltype_nargs():
+    '''Test that an exception is raised if the number of arguments does
+    not match the specified number of arguments in the kernel
+    metadata.
+
+    '''
+    my_code = CODE.replace("dimension(1)", "dimension(2)")
+    parse_tree = parse(my_code)
+
+    with pytest.raises(ParseError) as excinfo:
+        _ = KernelType(parse_tree)
+    assert ("In the 'meta_args' metadata, the number of args '2' and number "
+            "of dimensions '1' do not match.") in str(excinfo.value)
```

Comment:
It's not the number of dimensions - it's the extent of the dimension. Please could you amend the error message.

Reply:
Quite right, well noticed. Updated.